tests/drum/unit/test_typeschema_validation.py

```diff
@@ -692,3 +693,16 @@ def test_num_col_values(condition, value, fails):
             NumColumns(condition, value)
     else:
         NumColumns(condition, value)
+
+
+def test_datatypes_error_formatting():
```

Comment:
this should really test at a higher level and in the `TestSchemaValidator`

here's a copypaste test i took from: `tests.drum.unit.test_typeschema_validation.TestSchemaValidator.test_data_types`

```
    def test_data_types_error_message(
        self, ten_k_diabetes
    ):
        condition = Conditions.IN
        values = [Values.CAT, Values.NUM]
        target = "readmitted"

        yaml_str = input_requirements_yaml(Fields.DATA_TYPES, condition, values)
        schema_dict = self.yaml_str_to_schema_dict(yaml_str)
        validator = SchemaValidator(schema_dict)
        ten_k_diabetes.drop(target, inplace=True, axis=1)

        match_str = r'has types:( \w+,?){2,3}.* exactly match: CAT, NUM'
        with pytest.raises(DrumSchemaValidationException, match=match_str):
            validator.validate_inputs(ten_k_diabetes)
```